scripts/installer/install

```diff
@@ -180,8 +180,14 @@ if ( -e $config_default_file )
 		if ( 2 == @xs )
 		{
 			my $k = $xs[0];
-			my $v = trim($xs[1]);
-			$CONFIG_DEFAULTS{$k}=$v;
+                        my $v = trim($xs[1]);
+                        if (defined $ENV{$k} && ($k eq "POSTGRES_USER" || $k eq "POSTGRES_PASSWORD")) {
```

Comment:
If I'm reading this correctly, this changes the behavior of the install script to read from environmental variables instead of a configuration file and command line arguments.  Part of the thinking for using a configuration file was to defer the discussion of environmental variables vs command line arguments vs configuration files.

Options:
- accept additional tech debt for current installer (aka - config file, command line options and now environmental variables)
- discuss technical options, and decide one option for passing non-interactive options to the installer, and rework existing options to that choice.

For this PR, I think it makes sense to go with the first option.  Other thoughts?

Reply:
@pameyer this is exactly why I requested a code review from you. Is this going to break the ability to provide the config using a file?

Reply:
will need to test to confirm, but I don't believe it it will.

Reply:
@pameyer I just assigned this issue to myself and will probably test it in Vagrant.

@tkmonson is it clear what we're talking about? We don't want to lose the feature that @pameyer added in #3799 which is a way to run the installer non-interactively. The pull request was #3803.

Reply:
@pameyer I added a custom hostname in default.config (by temporarily editing install-dataverse.sh and I won't be checking in the change) and it seems to work fine. Please see below.

```
==> standalone: You should now have a running DVN instance at
==> standalone:   http://issue4763:8080
==> standalone: If vagrant up was successful (check output above) Dataverse is running on port 8080 of the Linux machine running within Vagrant, but this port has been forwarded to port 8888 of the computer you ran vagrant up on. For this reason you should go to http://localhost:8888 to see the Dataverse app running.
murphy:dataverse pdurbin$ 
murphy:dataverse pdurbin$ git diff
diff --git a/scripts/vagrant/install-dataverse.sh b/scripts/vagrant/install-dataverse.sh
index ac48217..0d4027c 100644
--- a/scripts/vagrant/install-dataverse.sh
+++ b/scripts/vagrant/install-dataverse.sh
@@ -19,7 +19,7 @@ if [ -e default.config ]; then
        mv default.config tmp-${pid}-default.config
 fi
 
-echo "HOST_DNS_ADDRESS localhost" > default.config
+echo "HOST_DNS_ADDRESS issue4763" > default.config
 echo "GLASSFISH_DIRECTORY      /home/glassfish/glassfish4" >> default.config
 
 if [ ! -z "$MAILSERVER" ]; then
murphy:dataverse pdurbin$ vagrant ssh
MAIL_SERVER environment variable not specified. Using localhost by default.
To specify it in bash: export MAIL_SERVER=localhost
[vagrant@standalone ~]$ sudo -i
[root@standalone ~]# grep issue4763 /home/glassfish/glassfish4/glassfish/domains/domain1/config/domain.xml
    <mail-resource host="localhost" from="do-not-reply@issue4763" user="dataversenotify" jndi-name="mail/notifyMailSession"></mail-resource>
        <jvm-options>-Ddataverse.fqdn=issue4763</jvm-options>
[root@standalone ~]# 
```

Reply:
@pdurbin This code still reads from the configuration file and writes the keys and values into the CONFIG_DEFAULTS hash. It just overrides the value in the config file if a corresponding environment variable has been defined (by openshift.json). So if a randomized user/password exists, its value is put into CONFIG_DEFAULTS instead of the default value.

The installer should be able to run non-interactively. That's how I used it in testing.